extensions/amp-story/1.0/test/test-amp-story.js

```diff
@@ -1198,6 +1199,147 @@ describes.realWin(
+            'amp-story-subscriptions'
+          );
+          story.element.appendChild(subscriptionsEl);
+
+          // buildCallback() could be omitted as long as createStoryWithPages()
+          // is called because the getImpl() call inside it.
+          await story.layoutCallback();
+        });
+
+        it('should continue navigating to locked pages after the subscription state gets resolved to granted', async () => {
+          // Block access with pending subscriptions state.
+          await story.switchTo_('page-1');
+          story.switchTo_('page-2');
+          expect(story.activePage_.element.id).to.equal('page-1');
+
+          // Once grant status resolves to true, goes back to normal story experiences.
+          story.storeService_.dispatch(
+            Action.TOGGLE_SUBSCRIPTIONS_STATE,
+            SubscriptionsState.GRANTED
+          );
+          await nextTick();
+          expect(story.activePage_.element.id).to.equal('page-2');
+
+          await story.switchTo_('page-3');
+          expect(story.activePage_.element.id).to.equal('page-3');
+        });
+
+        it('should block and switch dialog UI state after the subscription state gets resolved to blocked', async () => {
+          // Block access with pending subscriptions state.
+          await story.switchTo_('page-1');
+          story.switchTo_('page-2');
+          expect(story.activePage_.element.id).to.equal('page-1');
+
+          // Once grant status resolves to false, blocks the access to all locked pages and show the paywall.
+          // Note: would still go to the paywall page, e.g. page-2.
+          const dispatchSpy = env.sandbox.spy(story.storeService_, 'dispatch');
+          story.storeService_.dispatch(
+            Action.TOGGLE_SUBSCRIPTIONS_STATE,
+            SubscriptionsState.BLOCKED
+          );
+          await nextTick();
+          expect(story.activePage_.element.id).to.equal('page-2');
+
+          await setTimeout(() => {
+            // Should show the paywall.
+            expect(dispatchSpy).to.have.been.calledWith(
+              Action.TOGGLE_SUBSCRIPTIONS_DIALOG_UI_STATE,
+              true
+            );
+          }, PAYWALL_DELAY_DURATION);
+
+          await story.switchTo_('page-3');
+          expect(story.activePage_.element.id).to.equal('page-2');
+        });
+
+        it('should be able to navigate to next locked page once status becomes granted', async () => {
+          story.storeService_.dispatch(
+            Action.TOGGLE_SUBSCRIPTIONS_STATE,
+            SubscriptionsState.BLOCKED
+          );
+
+          await story.switchTo_('page-1');
+          await story.switchTo_('page-2');
+          await story.switchTo_('page-3');
+          expect(story.activePage_.element.id).to.equal('page-2');
+
+          // Should be able to navigate to next locked page once status becomes granted.
+          const dispatchSpy = env.sandbox.spy(story.storeService_, 'dispatch');
```

Comment:
you can get the store service without accessing private

Reply:
Done.